bigquery_etl/cli/backfill.py

```diff
@@ -401,12 +402,27 @@ def initiate(
     )
 
     # deploy backfill staging table
-    ctx.invoke(
-        deploy,
-        name=f"{dataset}.{table}",
-        project_id=project,
-        destination_table=backfill_staging_qualified_table_name,
-    )
+    try:
+        ctx.invoke(
+            deploy,
+            name=f"{project}.{dataset}.{table}",
+            project_id=project,
+            destination_table=backfill_staging_qualified_table_name,
+        )
+    except Exception as e:
+        raise Exception(
+            f"Deploying backfill staging table resulted in error for {qualified_table_name}."
+        ) from e
+        sys.exit(1)
```

Comment:
RE [this backfill failure](https://workflow.telemetry.mozilla.org/dags/bqetl_backfill_initiate/grid?dag_run_id=scheduled__2024-05-22T20%3A00%3A00%2B00%3A00&task_id=initiate_backfill.process_backfill&tab=mapped_tasks&map_index=0): I think the problem here is that the deploy didn't raise an Exception when it probably should have (since the deploy did technically fail, just silently). The error message: `No schema file found for ...` is a log/`click echo` message. 

I think the upstream code might need some changes to accommodate "skipped" deploys generally. Right now if there's no schema file that deploy is skipped but we still get an "all tables deployed" message.